sabnzbd/__init__.py

```diff
@@ -600,31 +601,34 @@ def backup_exists(filename: str) -> bool:
     return path and os.path.exists(os.path.join(path, filename + ".gz"))
 
 
-def backup_nzb(filename: str, data: AnyStr):
-    """Backup NZB file"""
-    path = cfg.nzb_backup_dir.get_path()
-    if path:
-        save_compressed(path, filename, data)
+def backup_nzb(nzb_path: str):
+    """Backup NZB file, return path to nzb if it was saved"""
+    nzb_backup_dir = cfg.nzb_backup_dir.get_path()
+    if nzb_backup_dir:
+        logging.debug("Saving copy of %s in %s", filesystem.get_filename(nzb_path), nzb_backup_dir)
+        shutil.copy(nzb_path, nzb_backup_dir)
 
 
-def save_compressed(folder: str, filename: str, data: AnyStr):
-    """Save compressed NZB file in folder"""
+def save_compressed(folder: str, filename: str, data: AnyStr) -> str:
+    """Save compressed NZB file in folder, return path to saved nzb file"""
     if filename.endswith(".nzb"):
         filename += ".gz"
     else:
         filename += ".nzb.gz"
-    logging.info("Backing up %s", os.path.join(folder, filename))
+    full_nzb_path = os.path.join(folder, filename)
+    logging.info("Saving %s", full_nzb_path)
     try:
         # Have to get around the path being put inside the tgz
-        with open(os.path.join(folder, filename), "wb") as tgz_file:
-            f = gzip.GzipFile(filename, fileobj=tgz_file, mode="wb")
-            f.write(encoding.utob(data))
-            f.flush()
-            f.close()
+        with open(full_nzb_path, "wb") as tgz_file:
+            # We only need minimal compression to prevent huge files
+            with gzip.GzipFile(filename, mode="wb", compresslevel=1, fileobj=tgz_file) as gzip_file:
+                gzip_file.write(encoding.utob(data))
```

Comment:
Opening tgz_file in a separate context manager before calling GzipFile isn't necessary: in case no fileobj is passed, the given filename is automatically opened to provide a file object.

Reply:
@jcfp No, otherwise it puts the whole filepath in the `tgz` if we don't do the double context manager.

Reply:
In that case, never mind...

Reply:
Isn't try and then two levels of with a bit excessive, though?

Reply:
Why? Each `with` serves the purpose of closing their respective file descriptors (error or no error) and the try/except serves the purpose of catching the error.

Reply:
I figured the first `with` would close the file anyway, but looking closer it's probably safer to do it this way.